src/article.ts

```diff
@@ -58,31 +58,42 @@ type Article
     | Standard
     ;
 
-type BodyElement = {
-    kind: ElementType.TEXT;
-    doc: DocumentFragment;
-} | {
-    kind: ElementType.IMAGE;
+const enum ElementKind {
+    Text,
+    Image,
+    Pullquote,
+    Interactive,
+    RichLink,
+    Tweet,
+}
+
+type Image = {
+    kind: ElementKind.Image;
     alt: string;
     caption: string;
     displayCredit: boolean;
     credit: string;
     file: string;
     width: number;
     height: number;
-} | {
-    kind: ElementType.PULLQUOTE;
+}
+
+type BodyElement = {
```

Comment:
While we're at it, should we split each `BodyElement` in separate types and have `type BodyElement= Image | Text | Interactive` etc?

Reply:
That seems sensible, it may let us narrow the types of functions further down the call stack.